src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -253,16 +264,50 @@ public DetailAST getUnderlyingNode() {
     }
 
     /**
-     * Finds child element with {@link TokenTypes#IDENT}, extracts its value and stores it.
-     * Value can be accessed using {@code @text} attribute. Now {@code @text} attribute is only
-     * supported attribute.
+     * Finds child element with {@link TokenTypes#IDENT},
+     * extracts its value, creates {@code AttributeNode} object and returns it.
+     * Value can be accessed using {@code @text} attribute.
+     *
+     * @return attribute object
      */
-    private void setIdent() {
-        final DetailAST identAst = detailAst.findFirstToken(TokenTypes.IDENT);
-        if (identAst != null) {
-            ident = identAst.getText();
+    private AbstractNode getIdent() {
+        AttributeNode attribute = null;
+        if (detailAst.getChildCount(TokenTypes.IDENT) == 1) {
+            ident = detailAst.findFirstToken(TokenTypes.IDENT).getText();
+            attribute = new AttributeNode(TEXT_ATTRIBUTE_NAME, ident);
+        }
+        return attribute;
+    }
+
+    /**
+     * Checks if current node is {@link TokenTypes#IDENT},
+     * extracts its value, creates {@code AttributeNode} object and returns it.
+     * Value can be accessed using {@code @value} attribute.
+     *
+     * @return attribute object
+     */
+    private AbstractNode getValue() {
+        AttributeNode attribute = null;
+        if (detailAst.getType() == TokenTypes.IDENT) {
```

Comment:
This limits the `value` attribute to `IDENT` nodes, solving #6115. Though all nodes contain any kind of text, which might be interesting. Think of XPaths just like `//NUM_INT[@value = 123]` or `//STRING_LITERAL[@value = '\"HelloWorld\"']`. Therefore I suggest leaving away this condition..

Reply:
done